tracer/src/Datadog.Trace/Iast/IastModule.cs

```diff
@@ -559,6 +561,12 @@ private static IastModuleResponse AddWebVulnerability(string? evidenceValue, Int
         if (!iastSettings.DeduplicationEnabled || HashBasedDeduplication.Instance.Add(vulnerability))
         {
             traceContext?.IastRequestContext?.AddVulnerability(vulnerability);
+
+            if (traceContext?.RootSpan is { } rootSpan)
+            {
+                rootSpan.Context.TraceContext?.SetSamplingPriority(SamplingPriorityValues.UserKeep, SamplingMechanism.Asm);
+            }
```

Comment:
This code is getting the root span from the trace context, only to go around and get the trace context from the root span. 🔁  Removing the `if`, it's basically doing:
```csharp
traceContext?.RootSpan?.Context.TraceContext?.SetSamplingPriority(...);
```

You can skip the root span entirely and do:
```suggestion
            traceContext?.SetSamplingPriority(SamplingPriorityValues.UserKeep, SamplingMechanism.Asm);
```

(You can also check if `traceContext` is `null` in the outer `if` (line 561) instead of checking twice with `traceContext?.`)

Reply:
Good catch! Thank you
For the nullability check, in fact, if the traceContext nullability is already checked on line 548.